src/TransientFETools/ODEOperatorInterfaces.jl

```diff
@@ -7,60 +7,90 @@ struct ODEOpFromFEOp{C} <: ODEOperator{C}
   feop::TransientFEOperator{C}
 end
 
+get_order(op::ODEOpFromFEOp) = get_order(op.feop)
+
 function allocate_cache(op::ODEOpFromFEOp)
   Ut = get_trial(op.feop)
-  Vt = ∂t(Ut)
   U = allocate_trial_space(Ut)
-  V = allocate_trial_space(Vt)
+  Uts = (Ut,)
+  Us = (U,)
+  for i in 1:get_order(op)
+    Uts = (Uts...,∂t(Uts[i]))
+    Us = (Us...,allocate_trial_space(Uts[i+1]))
+  end
   fecache = allocate_cache(op.feop)
-  ode_cache = (U,V,Ut,Vt,fecache)
+  ode_cache = (Us,Uts,fecache)
   ode_cache
 end
 
+function allocate_cache(op::ODEOpFromFEOp,v::AbstractVector,a::AbstractVector)
+  ode_cache = allocate_cache(op)
+  (v,a, ode_cache)
+end
+
 function update_cache!(ode_cache,op::ODEOpFromFEOp,t::Real)
-  U,V,Ut,Vt,fecache = ode_cache
-  U = evaluate!(U,Ut,t)
-  V = evaluate!(V,Vt,t)
+  _Us,Uts,fecache = ode_cache
+  Us = ()
+  for i in 1:get_order(op)+1
+    Us = (Us...,evaluate!(_Us[i],Uts[i],t))
+  end
   fecache = update_cache!(fecache,op.feop,t)
-  (U,V,Ut,Vt,fecache)
+  (Us,Uts,fecache)
 end
 
 function allocate_residual(op::ODEOpFromFEOp,uhF::AbstractVector,ode_cache)
-  U,V,Ut,Vt,fecache = ode_cache
-  uh = EvaluationFunction(U,uhF)
+  Us,Uts,fecache = ode_cache
+  uh = EvaluationFunction(Us[1],uhF)
   allocate_residual(op.feop,uh,fecache)
 end
 
 function allocate_jacobian(op::ODEOpFromFEOp,uhF::AbstractVector,ode_cache)
-  U,V,Ut,Vt,fecache = ode_cache
-  uh = EvaluationFunction(U,uhF)
+  Us,Uts,fecache = ode_cache
+  uh = EvaluationFunction(Us[1],uhF)
   allocate_jacobian(op.feop,uh,fecache)
 end
 
-function residual!(b::AbstractVector,op::ODEOpFromFEOp,t::Real,uhF::AbstractVector,uhtF::AbstractVector,ode_cache)
-  Uh,Uht, = ode_cache
-  uh = EvaluationFunction(Uh,uhF)
-  uht = EvaluationFunction(Uht,uhtF)
-  residual!(b,op.feop,t,uh,uht,ode_cache)
-end
-
-function jacobian!(A::AbstractMatrix,op::ODEOpFromFEOp,t::Real,uhF::AbstractVector,uhtF::AbstractVector,ode_cache)
-  Uh,Uht, = ode_cache
-  uh = EvaluationFunction(Uh,uhF)
-  uht = EvaluationFunction(Uht,uhtF)
-  jacobian!(A,op.feop,t,uh,uht,ode_cache)
+function residual!(
+  b::AbstractVector,
+  op::ODEOpFromFEOp,
+  t::Real,
+  xhF::Tuple{Vararg{AbstractVector}},
+  ode_cache)
+  Xh, = ode_cache
+  xh = ()
+  for i in 1:get_order(op)+1
+    xh = (xh...,EvaluationFunction(Xh[i],xhF[i]))
+  end
+  residual!(b,op.feop,t,xh,ode_cache)
 end
 
-function jacobian_t!(J::AbstractMatrix,op::ODEOpFromFEOp,t::Real,uhF::AbstractVector,uhtF::AbstractVector,dut_u::Real,ode_cache)
-  Uh,Uht, = ode_cache
-  uh = EvaluationFunction(Uh,uhF)
-  uht = EvaluationFunction(Uht,uhtF)
-  jacobian_t!(J,op.feop,t,uh,uht,dut_u,ode_cache)
+function jacobian!(
+  A::AbstractMatrix,
+  op::ODEOpFromFEOp,
+  t::Real,
+  xhF::Tuple{Vararg{AbstractVector}},
+  i::Integer,
+  γᵢ::Real,
+  ode_cache)
+  Xh, = ode_cache
+  xh = ()
+  for i in 1:get_order(op)+1
+    xh = (xh...,EvaluationFunction(Xh[i],xhF[i]))
+  end
+  jacobian!(A,op.feop,t,xh,i,γᵢ,ode_cache)
```

Comment:
I see that now we include the integer i to pick the Jacobian we want and a real number (also for i=1). We could probably put some documentation on the meaning of these things in case not already been done. It is a little bit messy to do the PR here with so many changes.

Reply:
It was done at the abstract method. I improved the documentation and added it to the ODEOperatorInterfaces.